dataservice.js

```diff
@@ -0,0 +1,46 @@
+const categoriesJson = require('./my-app-react/src/data/categories.json');
+const productsJson = require('./my-app-react/src/data/products.json');
+
+let combinedProductMap = new Map();
+let productMap = new Map();
+let categoryMap = new Map();
+
+let getProducts = () => {
+    for (let product of productsJson.products) {
```

Comment:
If productMap size is zero, then load the JSON from file, and loop it to create new Map.
Otherwise, just return productMap